src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -307,6 +307,20 @@ private static bool IsFloatPointExpressionOfUnknownPrecision(BoundExpression rew
 
                     break;
 
+                case ConversionKind.MethodGroup:
+
+                    if (_inExpressionLambda || explicitCastInCode || (object)symbolOpt == null || !symbolOpt.IsStatic || symbolOpt.MethodKind != MethodKind.Ordinary)
+                    {
+                        break;
+                    }
+                    if (symbolOpt.IsExtensionMethod && ((BoundMethodGroup)rewrittenOperand).ReceiverOpt.Kind != BoundKind.TypeExpression)
```

Comment:
The check you're trying to do is test if the method is an extension method being accessed as an instance method. We should consider factoring out a helper here, if one doesn't already exist.